src/widget-context-menu/widget-context-menu.ts

```diff
@@ -0,0 +1,75 @@
+import type { ChildClient } from '@datadog/framepost';
+
+import { UiAppFeatureType, UiAppRequestType } from '../constants';
+import { DDFeatureClient } from '../shared/feature-client';
+import type {
+    GetWidgetContextMenuItemsRequest,
+    GetWidgetContextMenuItemsResponse
+} from '../types';
+import type { Logger } from '../utils/logger';
+import { validateKey } from '../utils/utils';
+
+const emptyConfig: GetWidgetContextMenuItemsResponse = { items: [] };
+
+export class DDWidgetContextMenuClient extends DDFeatureClient {
+    constructor(debug: boolean, logger: Logger, framePostClient: ChildClient) {
+        super(
+            debug,
+            logger,
+            framePostClient,
+            UiAppFeatureType.WIDGET_CONTEXT_MENU
+        );
+
+        // initialize with an empty reponse handler
+        this.onRequestItems(() => emptyConfig);
+    }
+
+    /**
+     * Registers a request handler for providing context menu items dynamically
+     */
+    onRequestItems(
```

Comment:
while technically `onRequestItems` makes perfect sense to describe what this method does, I think either replacing it or wrapping it with something more generic like `init` make more sense and could be more intuitive to users. This allows us to future proof it so it be reused it if we decide to allow users to define other properties in addition to items like titles for example. We could also use the same name for more features in the future that are not necessarily items. What do you think?

Reply:
We have a pattern already of prefixing with `on` when registering event or request handlers, so i think we should use `on`, but I think renaming it to `onRequest` makes sense and is shorter. Does that make sense to you?

Reply:
yeah sure, i think that's a good middle ground :)